### ISSUE TITLE
Question about gap between err_tr and err_te

### ISSUE THREAD

**Issue by JiachuanDENG** [mravanelli/SincNet#14]
I really appreciate you released the paper together with the source code. Also, I have a question on the performance gap between err_tr and err_te:

According to the result shown in res.res, after epoch 360, err_tr=0.009600, err_te=0.419954. The gap between training and validation's performance seems quite large. Does it mean the model suffers some kind of overfitting problem?

**Comment by mravanelli** [comment]
Yes,
the model tends to overfit (especially with small dataset like TIIMIT). We
tried higher dropout factors to regularize a bit more the classifier, but
the best performance on the test set is obtained with the current
configuration.

**Comment by mravanelli** [comment]
No,
the performance reported in the paper is on the test set and is about
sentence-level classification.

Let me clarify how one should interpret the res.res file:

epoch 360, loss_tr=0.033095 err_tr=0.009600 loss_te=4.254683
err_te=0.419954 err_te_snt=0.005772

"err_tr" is the frame error rate on the training dataset. The frame error
rate is computed at each frame of 200ms and is different from the sentence
error rate reported in the paper.
Similarly, the "err_te" contains the *frame* error rate. The results
reported in the paper are the sentence-level error rate, that is reported
in the field "err_te_snt". The sentence-level error rate (which the metric
that really matters) is computed by averaging the posterior probabilities
computed at each frame composing the sentence and voting for the speaker
with the highest average probability.  In the example reported in the paper
you reached a CER=0.57%. In the paper, it is reported a CER=0.85% because
to obtained more stable and reliable numbers we run the experiment multiple
times with different initialization seeds and we reported the average CER.

Please, let me know if this clarifies your doubts.

Best,

Mirco

**Comment by JiachuanDENG** [comment]
Thanks for your clarification. I have no doubt on this question now :)
